packages/utils/telemetry-utils/src/logger.ts

```diff
@@ -150,7 +156,7 @@ export abstract class TelemetryLogger implements ITelemetryLogger {
         if (typeof newEvent.duration === "number") {
             newEvent.duration = TelemetryLogger.formatTick(newEvent.duration);
         }
-
+        convertToBaseEvent(newEvent);
```

Comment:
I still think this function should return a new object rather than trying to modify the one passed in. You can replace `const newEvent = { ...event };` with `const newEvent = convertToBaseEvent(event);`.

Reply:
On a related note, let's try to leave the error telemetry props using the base types, since that interface is exposed on `IErrorBase` which is a public interface.  So `prepareErrorObject` function will remain unchanged.